### PR TITLE
feat: ✨ added includeQueryParams flag

### PR DESCRIPTION
**Pull request by dragos199993**
### Motivation and Context

Gives the option to use the entire URL in order to have a cache key that includes query parameters.
Fixes: https://github.com/bubkoo/html-to-image/issues/259

### Types of changes

- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] Enhancement (changes that improvement of current feature or performance)
- [ ] Refactoring (changes that neither fixes a bug nor adds a feature)
- [ ] Test Case (changes that add missing tests or correct existing tests)
- [ ] Code style optimization (changes that do not affect the meaning of the code)
- [ ] Docs (changes that only update documentation)
- [ ] Chore (changes that don't modify src or test files)

### Self Check before Merge

- [x] My code follows the code style of this project.
- [ ] My change requires a change to the documentation.
- [x] I have updated the documentation accordingly.
- [x] I have read the [**CONTRIBUTING**](https://github.com/bubkoo/html-to-image/blob/master/CONTRIBUTING.md) document.
- [ ] I have added tests to cover my changes.
- [x] All new and existing tests passed.

### CODE REVIEW

File: src/embedWebFonts.ts [hunk 1]
```diff
@@ -141,7 +141,9 @@ async function getCSSRules(
                         rule,
                         rule.startsWith('@import')
                           ? (importIndex += 1)
-                          : sheet.hasOwnProperty('cssRules').length,
+                          : // eslint-disable-next-line @typescript-eslint/ban-ts-comment
+                            // @ts-ignore
+                            sheet.hasOwnProperty('cssRules').length,
```
Comment (dragos199993) [inline src/embedWebFonts.ts:146]: Is there any other solution to this? `hasOwnProperty` returns boolean so `.length` is not a valid method

**Review by emanuel-pislea — APPROVED** [review]

**Review by galoberlyn — APPROVED** [review]

File: src/embedWebFonts.ts (hunk 1 above)
Comment (ryan3sg) [reply src/embedWebFonts.ts:146]: can we use `('cssRules' in sheet).length`for this?

**Comment by dragos199993** [comment]
This PR is still valid

**Comment by MatanYemini** [comment]
Any reason why not move this forward?

**Comment by osplm** [comment]
Was about to open a PR exactly for this problem. Would be nice to merge this 👍

File: src/getBlobFromURL.ts [hunk 2]
```diff
@@ -10,10 +10,14 @@ const cache: {
   [url: string]: Promise<Metadata>
 } = {}
 
-function getCacheKey(url: string) {
+function getCacheKey(url: string, includeQueryParams: boolean | undefined) {
   let key = url.replace(/\?.*/, '')
```
Comment (osplm) [inline src/getBlobFromURL.ts:14]: Nitpick: No need to replace if we will reassign the full url after so we should replace only if it's not to includeQueyParams right?

**Comment by ortonomy** [comment]
Really need this. This lib is incompatible with libraries like next.js which caches images and retrieves them like `/next/_image?url=.....` -- which appear to `html-to-image` as the same URL currently

**Comment by RodrigoTomeES** [comment]
Hi, any update about this?

**Comment by ortonomy** [comment]
For anyone else looking for a solution to this, at least with next.js, I found out that next.js has a built in proxy that they call 'redirects' -- I was able to make it look like the image was coming from local, and with a different URL each time by not using next.js's built in image lib and proxying requests so they were 'secure' via the server: https://nextjs.org/docs/api-reference/next.config.js/redirects

**Comment by RodrigoTomeES** [comment]
There is any plan to merge this PR?

**Pull request merged**